src/backend-test/index.ts

```diff
@@ -0,0 +1,236 @@
+      srcdir: options.srcdir ?? '.',
+
+      tsconfig: {
+        compilerOptions: {
+          declaration: undefined,
+          experimentalDecorators: undefined,
+          inlineSourceMap: undefined,
+          inlineSources: undefined,
+          noEmitOnError: undefined,
+          noUnusedLocals: undefined,
+          lib: undefined,
+          outDir: undefined,
+          rootDir: undefined,
+          stripInternal: undefined,
+          forceConsistentCasingInFileNames: true,
+          isolatedModules: false,
+          module: 'esnext',
+          moduleResolution: TypeScriptModuleResolution.NODE,
+          noEmit: true,
+          noUncheckedIndexedAccess: true,
+          baseUrl: './',
+          target: 'es6',
+          skipLibCheck: true,
+          paths: {
+            '*': ['./*'],
+          },
+        },
+      },
+
+      // In case Github is enabled remove all default stuff.
+      githubOptions: {mergify: false, pullRequestLint: false},
+      buildWorkflow: false,
+      release: false,
+      depsUpgrade: false,
+      pullRequestTemplate: false,
+      deps: [
+        'dotenv',
+        'axios',
+        'jest',
+        'ts-jest',
+        'jest-extended',
+        'prettier',
+        'eslint-plugin-prettier',
+        'eslint-config-prettier',
+      ],
+    })
+
+    // ANCHOR Source code
+    const assetsDir = path.join(__dirname, '..', '..', 'src/backend-test/assets')
+    sampleCode(this, options, assetsDir)
+
+    /*
+     * Clean off the projen tasks and if needed replace them with regular npm scripts.
+     * This way we ensure smooth ejection experience with all the commands visible in package.json
+     * and no need to keep the projen task runner within an ejected project.
+     */
+    const tasksToRemove = [
+      'build',
+      'bump',
+      'clobber',
+      'compile',
+      'package',
+      'post-compile',
+      'post-upgrade',
+      'pre-compile',
+      'release',
+      'test',
+      // eslint-disable-next-line @cspell/spellchecker -- the word is used once, so no need to add it to the dictionary
+      'unbump',
+      'upgrade',
+      'watch',
+      'projen',
+    ]
+
+    tasksToRemove.forEach(this.removeTask.bind(this))
+
+    this.tryRemoveFile('tsconfig.dev.json')
+
+    // ANCHOR Jest setup
+    this.addScripts({
+      test: 'jest --detectOpenHandles',
+    })
+
+    new AssetFile(this, 'jest.config.ts', {
+      sourcePath: path.join(assetsDir, 'jest.config.ts'),
+      readonly: false,
+      marker: false,
+    })
+
+    // ANCHOR ESLint and prettier setup
+    new JsonFile(this, '.eslintrc.json', {obj: eslintConfig, marker: false})
+
+    new JsonFile(this, '.prettierrc.json', {obj: prettierConfig, marker: false})
+
+    new AssetFile(this, '.prettierignore', {
+      sourcePath: path.join(assetsDir, '.prettierignore'),
+      readonly: false,
+      marker: false,
+    })
+
+    this.tasks.addTask('format', {
```

Comment:
I need we need a description why you abandoned the common `addLInters` function and try to replicate it manually. So far I see no reason for this.

Reply:
Based on your suggestion I used the addLinters function for passing .eslintrc. Unfortunately, I still rewrite tasks such as 'format' etc with my custom
<img width="446" alt="Screenshot 2023-12-21 at 16 04 48" src="https://github.com/ottofeller/templates/assets/130353062/ebc1a3d5-5e25-4139-b9fc-86bf36ef39cd">
 command cause have a problem with file paths

Reply:
`addLinters` has a parameter `lintPaths` for this case. I still don't see why you might need to overwrite something. From what I see, you just need to provide `lintPaths: ['.']`.

However if something finally does not solve your problem, let's discuss it and enhance the `addLinters` function. Otherwise it looks very weird when we use escape hatches even inside this repo.

Reply:
yep, I did that on line 147. But in my previous comment, I talked about scripts which Projen generate for package.json. If I use the default one I get an error which is why I rewrite scripts like 'format', 'lint' and 'typecheck'. Of course, I can do this in projenrc.ts file when creating the project if it is better

Reply:
The error in [your comment](https://github.com/ottofeller/templates/pull/475#discussion_r1434191930) is self-explanatory - you tried to check `src` folder, which does not exist. With the updated `lintPaths` you don't get it and hence there is no need to overwrite the tasks. Neither you need to do it in the `projenrc.ts` file - that would be an anti pattern that we better solve at the template level.